### PR TITLE
Ads control

### PR DESCRIPTION
**Pull request by globaevg**

### CODE REVIEW

File: client/src/common/carousel/Carousel.jsx [hunk 1]
```diff
@@ -5,7 +5,6 @@ import { Carousel } from 'react-responsive-carousel';
 import './carousel.scss';
 
 const Carouseles = ({ images = images.images }) => {
-  console.log(images.length);
   return (
     <Carousel emulateTouch className='carousel-flat-img'>
       {images.length == 0 ? (
```
Comment (antonstan) [inline client/src/common/carousel/Carousel.jsx]: how we can replace it with?

File: client/src/components/Flat.jsx [hunk 2]
```diff
@@ -1,8 +1,18 @@
-import React from 'react';
+import React, { Fragment } from 'react';
 import Carouseles from 'common/carousel/Carousel';
+import Button from 'common/button/Button';
 import './flat.scss';
+// import { getFlat } from '../../../server/middleware/flats';
 
 const Flat = attrs => {
```
Comment (antonstan) [inline client/src/components/Flat.jsx]: concept attrs not applicable for the props 
it is inconsistent

Comment (antonstan) [reply client/src/components/Flat.jsx]: use destructuration instead of

File: client/src/components/Flat.jsx [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
+  const deleteAd = () => {
+    const { deleteFlat } = attrs;
```
Comment (antonstan) [inline client/src/components/Flat.jsx]: please apply destructuration for props

Comment (antonstan) [reply client/src/components/Flat.jsx]: and below

File: client/src/components/pages/addFlat/AddFlat.jsx [hunk 4]
```diff
@@ -38,23 +31,11 @@ class AddFlat extends Component {
...
-      flat_number: this.state.flat_number,
-      flore_number: this.state.flore_number,
-      square_meters_number: this.state.square_meters_number,
-      price_per_month: this.state.price_per_month,
-      discription: this.state.discription,
-      pictures: this.state.pictures
+      ...this.state
     };
-
     addflat(dataflat);
```
Comment (antonstan) [inline client/src/components/pages/addFlat/AddFlat.jsx]: use camelCase

File: client/src/components/Flat.jsx [hunk 5]
```diff
@@ -16,7 +26,21 @@ const Flat = attrs => {
...
+              <Fragment>
+                <Button onClick={deleteAd}>Delete</Button>
+                <Button onClick={getOneFlat} to={`/my-profile/${attrs._id}`}>
+                  Adjustment
+                </Button>
+              </Fragment>
+            )}
+          </Fragment>
+        ) : (
+          ''
```
Comment (antonstan) [inline client/src/components/Flat.jsx]: null

File: client/src/components/pages/addFlat/AddFlat.jsx (hunk 4 above)
Comment (antonstan) [inline client/src/components/pages/addFlat/AddFlat.jsx]: addFlat(this.state) ?

File: client/src/components/pages/addFlat/AddFlat.jsx [hunk 6]
```diff
@@ -150,12 +130,12 @@ class AddFlat extends Component {
             minLength='40'
```
Comment (antonstan) [inline client/src/components/pages/addFlat/AddFlat.jsx]: why do you send numbers as a string?

File: client/src/components/pages/adsAdjustment/AdsAdjustment.jsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    changeFlat(dataRegister);
+  };
+
+  render() {
+    const { flat } = this.props;
+    return (
+      <div>
+        {flat !== null && (
```
Comment (antonstan) [inline client/src/components/pages/adsAdjustment/AdsAdjustment.jsx]: how we can replace it?

File: client/src/components/pages/adsAdjustment/AdsAdjustment.jsx [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+                label='select number'
+                type='number'
+                name='house_number'
+                defaultValue={flat.house_number}
+                onChange={this.onChange}
+              />
+
+              <h2>Enter number of corps:</h2>
+              <Input
+                id='number'
```
Comment (antonstan) [inline client/src/components/pages/adsAdjustment/AdsAdjustment.jsx]: please do not use the same ids for inputs you are breaking dom logic. each element should has unique id

Comment (antonstan) [reply client/src/components/pages/adsAdjustment/AdsAdjustment.jsx]: please fix it anywhere

File: client/src/components/pages/myProfile/MyProfile.jsx [hunk 9]
```diff
@@ -71,6 +78,16 @@ class MyProfile extends Component {
           <span />
           <Button onClick={this.onSubmit}>submit</Button>
         </div>
+        {flats.length == 0 || flats.length == undefined ? (
```
Comment (antonstan) [inline client/src/components/pages/myProfile/MyProfile.jsx]: how we can replace it?

File: client/src/router/SiteRouter.jsx [hunk 10]
```diff
@@ -14,6 +15,7 @@ const SiteRouter = () => {
       <Route exact path='/flats' component={Flats} />
       <PrivateRoute exact path='/add-flat' component={AddFlat} />
       <PrivateRoute exact path='/my-profile' component={MyProfile} />
+      <PrivateRoute exact path='/my-profile/:id' component={AdsAdjustment} />
```
Comment (antonstan) [inline client/src/router/SiteRouter.jsx]: this route hasn't be exact

File: server/middleware/auth.js [hunk 11]
```diff
@@ -5,7 +5,7 @@ require('dotenv').config();
 var HttpStatus = require('http-status-codes');
 const { findUser, getUserWithoutPassword } = require('../repositories/user');
 
-exports.checkauth = function(req, res, next) {
+exports.checkauth = (req, res, next) => {
```
Comment (antonstan) [inline server/middleware/auth.js]: camelCase please

File: server/middleware/cloudinary.js [hunk 12]
```diff
@@ -21,12 +20,12 @@ const storage = cloudinaryStorage({
...
 });
 
-exports.uploadFlat = (req, res) => {
+exports.imgHandler = (req, res, next) => {
   const upload = multer({
     storage: storage
   }).any();
 
-  upload(req, res, function(err) {
+  upload(req, res, async err => {
```
Comment (antonstan) [inline server/middleware/cloudinary.js]: why callback is async ? 
does this function have promise variant?

File: server/middleware/flats.js [hunk 13]
```diff
@@ -12,44 +18,86 @@ module.exports.getFlats = async (req, res) => {
...
+    const flat = await getFlatQuery(req.params.id);
+    res.json(flat);
+  } catch (err) {
+    res.status(HttpStatus.BAD_REQUEST);
+  }
+};
+
+module.exports.addFlat = async (req, res) => {
+  try {
+    const flatFields = ({
```
Comment (antonstan) [inline server/middleware/flats.js]:
```suggestion
    const flatFields = ({...req.data.body})
``` ?

File: server/middleware/flats.js [hunk 14]
```diff
@@ -12,44 +18,86 @@ module.exports.getFlats = async (req, res) => {
...
+      msg: 'Flat removed'
+    });
+  } catch (err) {
+    res.status(HttpStatus.BAD_REQUEST);
+  }
+};
+
+module.exports.changeFlat = async (req, res) => {
+  try {
+    const flatFields = ({
```
Comment (antonstan) [inline server/middleware/flats.js]: the same

File: server/models/User.js [hunk 15]
```diff
@@ -23,7 +24,13 @@ const UserSchema = new mongoose.Schema({
   date: {
     type: Date,
     default: Date.now
-  }
+  },
+  ads: [
```
Comment (antonstan) [inline server/models/User.js]: Please read about mongo aggregation and lookup and about indexes and make changes

File: server/repositories/flat.js [hunk 16]
```diff
@@ -78,3 +87,50 @@ exports.deleteFlats = async () => {
...
+
+    return flat;
+  } catch (err) {
+    console.log('erron inside of saveFlat ' + err);
+  }
+};
+
+exports.getFlatsUserQuery = async params => {
+  try {
+    const flatsUserQuery = await Flat.find({ _id: { $in: params } });
```
Comment (antonstan) [inline server/repositories/flat.js]: mongo aggregation and lookup

File: client/src/components/pages/addFlat/AddFlat.jsx (hunk 6 above)
Comment (globaevg) [reply client/src/components/pages/addFlat/AddFlat.jsx]: if like a number, it doesn't work

File: server/middleware/cloudinary.js (hunk 12 above)
Comment (globaevg) [reply server/middleware/cloudinary.js]: I don't know how it appeared here. By mistake

File: client/src/common/carousel/Carousel.jsx (hunk 1 above)
Comment (globaevg) [reply client/src/common/carousel/Carousel.jsx]: images.length == 0, if about it, we can't. I tried everything what I could.

File: client/src/components/pages/myProfile/MyProfile.jsx [hunk 17]
```diff
@@ -78,7 +77,7 @@ class MyProfile extends Component {
           <span />
           <Button onClick={this.onSubmit}>submit</Button>
         </div>
-        {flats.length == 0 || flats.length == undefined ? (
+        {flats.length == undefined ? (
```
Comment (antonstan) [inline client/src/components/pages/myProfile/MyProfile.jsx]: !flats.length

File: client/src/components/pages/adsAdjustment/AdsAdjustment.jsx [hunk 18]
```diff
@@ -85,7 +86,7 @@ class AdsAdjustment extends Component {
 
               <h2>Enter number of flat:</h2>
               <Input
-                id='number'
+                id={uuidv1()}
```
Comment (antonstan) [inline client/src/components/pages/adsAdjustment/AdsAdjustment.jsx]: not a variant. id will change per each rerender 
please use unique id like flat-number-inp

Comment (globaevg) [reply client/src/components/pages/adsAdjustment/AdsAdjustment.jsx]: got it

**Pull request merged**